deepspeed/runtime/engine.py

```diff
@@ -1137,10 +1137,6 @@ def _do_optimizer_sanity_check(self, basic_optimizer):
 
                 if self.global_rank == 0:
                     logger.warning("**** You are using ZeRO with an untested optimizer, proceed with caution *****")
-
-            if model_dtype == torch.bfloat16 and grad_accum_dtype == torch.float32 and self.zero_optimization_stage(
```

Comment:
This change could break backwards-compatibility for users who expect to use bf16_optimizer in this case.